test/unit/prof_use_sys_thread_name.c

```diff
@@ -0,0 +1,75 @@
+#include "test/jemalloc_test.h"
+
+static const char *test_thread_name = "test_name";
+
+static int
+test_prof_read_sys_thread_name_error(char *buf, size_t limit) {
+	return -1;
+}
+
+static int
+test_prof_read_sys_thread_name(char *buf, size_t limit) {
+	assert(strlen(test_thread_name) < limit);
+	strcpy(buf, test_thread_name);
```

Comment:
A raw strcpy will trigger warnings in some of our CI setups IIRC. Probably better is strncpy, even with the assert.